apps/prairielearn/src/lib/question.js

```diff
@@ -1299,14 +1330,40 @@ export async function startTestQuestion(
         course,
         type,
         authn_user_id,
-        (err, ret_success) => {
-          if (ERR(err, callback)) return;
-          success = success && ret_success;
-          callback(null);
-        },
       );
+      success = success && result.success;
+      if (result.stats) {
+        stats.push(result.stats);
+      }
     });
 
+    function printStats(label, key) {
+      let min = Number.MAX_SAFE_INTEGER;
+      let max = 0;
+      let count = 0;
+      let sum = 0;
+      stats.forEach((stat) => {
+        const value = stat[key];
+        if (value == null) return;
+        count += 1;
+        sum += value;
+        min = Math.min(min, value);
+        max = Math.max(max, value);
+      });
+
+      if (count === 0) {
+        job.verbose(`${label}: No data`);
+        return;
+      }
+
+      const avg = Math.round((sum / count) * 100) / 100;
+      job.info(`${label}: ${count} tests, min ${min}ms, max ${max}ms, avg ${avg}ms`);
+    }
+
+    printStats('Generate/prepare', 'generateDuration');
+    printStats('Render', 'renderDuration');
+    printStats('Parse/grade', 'gradeDuration');
```

Comment:
I think aligning them would make it easier to read the results, even if it's just the labels:
```suggestion
    printStats('Generate/prepare', 'generateDuration');
    printStats('Render          ', 'renderDuration');
    printStats('Parse/grade     ', 'gradeDuration');
```

Reply:
Addressed in 64cac25.